java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/DbzConnectorConfig.java

```diff
@@ -354,4 +369,76 @@ private Properties initiateDbConfig(String fileName, StringSubstitutor substitut
         }
         return dbProps;
     }
+
+    /**
+     * Calculate and set max.queue.size.in.bytes based on JVM heap size and memory ratio.
+     *
+     * <p>If user does not specify queue.memory.ratio, this method does nothing. Debezium will use
+     * whatever is configured in debezium.properties (max.queue.size = 8192 by default).
+     *
+     * <p>If user specifies a valid ratio (0.01-0.8), calculate and set max.queue.size.in.bytes
+     * accordingly.
+     *
+     * <p>JVM heap size is obtained from Runtime.getRuntime().maxMemory(), which reflects the -Xmx
+     * value set by Rust code during JVM initialization. See: src/jni_core/src/jvm_runtime.rs:85-103
+     *
+     * <p>The Rust code sets -Xmx based on: 1. JVM_HEAP_SIZE env var if set 2. Otherwise:
+     * system_memory_available_bytes() * 0.07
+     *
+     * @param dbzProps The Debezium properties to update
+     */
+    private void calculateAndSetMaxQueueSizeInBytes(Properties dbzProps) {
+        // Check if user specified a custom ratio
+        String ratioStr = dbzProps.getProperty(QUEUE_MAX_MEMORY_RATIO);
+        if (ratioStr == null) {
+            // User did not specify ratio, do nothing
+            // Debezium will use its default max.queue.size from debezium.properties
+            LOG.info(
+                    "Debezium {} not specified, skipping calculating and setting max.queue.size.in.bytes calculation",
+                    QUEUE_MAX_MEMORY_RATIO);
+            return;
+        }
+
+        double queueMemoryRatio;
+        try {
+            queueMemoryRatio = Double.parseDouble(ratioStr);
+            // Validate ratio is in reasonable range (1% - 80%)
+            if (queueMemoryRatio < MIN_QUEUE_MAX_MEMORY_RATIO
+                    || queueMemoryRatio > MAX_QUEUE_MAX_MEMORY_RATIO) {
+                LOG.warn(
+                        "Invalid {}: {}, must be between {} and {}, ignoring",
+                        QUEUE_MAX_MEMORY_RATIO,
+                        queueMemoryRatio,
+                        MIN_QUEUE_MAX_MEMORY_RATIO,
+                        MAX_QUEUE_MAX_MEMORY_RATIO);
+                return;
```

Comment:
Can we validate this during `CREATE SOURCE` and `ALTER SOURCE s CONNECTOR WITH` and throw an SQL error if user inputs a bad ratio?